util/resolver/authorizer.go

```diff
@@ -400,6 +400,11 @@ func (ah *authHandler) fetchToken(ctx context.Context, sm *session.Manager, g se
 }
 
 func invalidAuthorization(c auth.Challenge, responses []*http.Response) error {
+	lastResponse := responses[len(responses)-1]
+	if lastResponse.StatusCode >= http.StatusBadRequest {
```

Comment:
why is this `4**` . I thought the point was to only check `StatusUnauthorized`

Reply:
It seemed like other errors should also imply that the token was bad. I can change it to narrow it down if you like.

Reply:
Done.